sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -0,0 +1,555 @@
+package com.horizen.account.api.rpc.service
+
+import akka.actor.ActorRef
+import akka.pattern.ask
+import akka.util.Timeout
+import com.horizen.SidechainSettings
+import com.horizen.account.api.rpc.handler.RpcException
+import com.horizen.account.api.rpc.types._
+import com.horizen.account.api.rpc.utils._
+import com.horizen.account.block.AccountBlock
+import com.horizen.account.history.AccountHistory
+import com.horizen.account.mempool.AccountMemoryPool
+import com.horizen.account.secret.PrivateKeySecp256k1
+import com.horizen.account.state._
+import com.horizen.account.transaction.EthereumTransaction
+import com.horizen.account.utils.AccountForwardTransfersHelper.getForwardTransfersForBlock
+import com.horizen.account.utils.EthereumTransactionDecoder
+import com.horizen.account.wallet.AccountWallet
+import com.horizen.api.http.SidechainTransactionActor.ReceivableMessages.BroadcastTransaction
+import com.horizen.chain.SidechainBlockInfo
+import com.horizen.evm.interop.TraceParams
+import com.horizen.evm.utils.{Address, Hash}
+import com.horizen.params.NetworkParams
+import com.horizen.transaction.exception.TransactionSemanticValidityException
+import com.horizen.utils.{ClosableResourceHandler, TimeToEpochUtils}
+import org.web3j.crypto.Sign.SignatureData
+import org.web3j.crypto.{SignedRawTransaction, TransactionEncoder}
+import org.web3j.utils.Numeric
+import scorex.util.{ModifierId, ScorexLogging}
+import sparkz.core.NodeViewHolder.CurrentView
+import sparkz.core.{NodeViewHolder, bytesToId}
+
+import java.math.BigInteger
+import java.util
+import scala.collection.JavaConverters.seqAsJavaListConverter
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+import scala.concurrent.duration.{FiniteDuration, SECONDS}
+import scala.concurrent.{Await, Future}
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+class EthService(
+    val scNodeViewHolderRef: ActorRef,
+    val nvtimeout: FiniteDuration,
+    networkParams: NetworkParams,
+    val sidechainSettings: SidechainSettings,
+    val sidechainTransactionActorRef: ActorRef
+) extends RpcService
+      with ClosableResourceHandler
+      with ScorexLogging {
+  type NV = CurrentView[AccountHistory, AccountState, AccountWallet, AccountMemoryPool]
+
+  def applyOnAccountView[R](functionToBeApplied: NV => R): R = {
+    implicit val timeout: Timeout = new Timeout(nvtimeout)
+    val res = scNodeViewHolderRef
+      .ask {
+        NodeViewHolder.ReceivableMessages.GetDataFromCurrentView { (nodeview: NV) =>
+          // wrap any exceptions
+          Try(functionToBeApplied(nodeview))
+        }
+      }
+      .asInstanceOf[Future[Try[R]]]
+    // return result or rethrow potential exceptions
+    Await.result(res, nvtimeout) match {
+      case Success(value) => value
+      case Failure(exception) =>
+        exception match {
+          case err: RpcException => throw err
```

Comment:
The RPC error handling could use a refactoring and error responses need to be checked against the scpecs.